### ISSUE TITLE
[Bug]: Strategy: "apiToken"

### ISSUE THREAD

**Issue by foodiecompany** [ComfortablyCoding/strapi-plugin-io#86]
### What issue are you experiencing?

The apiToken strategy encounters issues when modifying specific data, and the cause of this problem is currently unknown. However, a workaround is available by switching to the JWT strategy, which allows the messages for changes to function as expected. Your assistance in resolving this matter would be greatly appreciated. Thank you.

#### Plugin config

```ts
const config = {
  io: {
    enabled: true,
    config: {
      socket: {
        serverOptions: {
          cors: {
            origin: [env("CLIENT_URL"), env("STAFF_URL")],
            methods: ["GET", "POST"],
            credentials: true,
          },
        },
      },
      contentTypes: [
        "api::table.table",
        "api::request-staff-event.request-staff-event",
        "api::order.order",
        "api::cart.cart",
      ],
      events: [
        {
          name: "connection",
          handler: ({ strapi }, socket) => {
            strapi.log.info(`[io] new connection with id ${socket.id}`);
          },
        },
      ],
    },
  },

}
```

#### Token Configuration
![image](https://github.com/ComfortablyCoding/strapi-plugin-io/assets/146194261/3158c12a-8091-467c-9488-9fefa55d547f)

### FrontEnd NextJS
```ts
const so = io(config.apiUrl, {
    withCredentials: true,
    auth: {
      strategy: "apiToken",
      token: getConfig().guestToken
    }
  });
```

### Networking

Payload
![image](https://github.com/ComfortablyCoding/strapi-plugin-io/assets/146194261/9fd43062-cfa8-4293-8792-80159f785183)

Responce
![image](https://github.com/ComfortablyCoding/strapi-plugin-io/assets/146194261/8f5d9156-0210-42e5-b13f-49d3542b1464)

Comunications
![image](https://github.com/ComfortablyCoding/strapi-plugin-io/assets/146194261/562ec8ab-b98f-4d65-ac67-d3e69a6b88e4)

### What version of the plugin are you using?

2.0.5

### What strapi version are you using?

4.17.1

**Comment by ComfortablyCoding** [comment]
Thanks for the report! Can  you expand on what you mean by "The apiToken strategy encounters issues when *modifying specific data*"? The reproduction steps are not so clear to me. 

How are you sending the data? What are you expecting to receive back vs what you actually receiving? Is their any custom function that is handling it server side? Any other possible relevant info?

**Comment by foodiecompany** [comment]
Sure,
When the auth type `jwt` all works fine but, when I change to `apiToken` does.

As previously mentioned, we have a collectionType: api::table.table, containing fields like no, descriptions, status, etc.. Everything functions correctly when modifying the 'status' field under the 'jwt' strategy. However, issues arise when using the 'apiToken' strategy.

**Comment by ComfortablyCoding** [comment]
What "issues" would that be?

**Comment by foodiecompany** [comment]
Our attached handler is not being called

Example:
```ts
socket.on("table:update", (table) => {

});
```

The event `table:update` is not triggered when we use the `apiToken` strategy.

**Comment by ComfortablyCoding** [comment]
Thank you for the clarification, I will see if I can reproduce this issue

**Comment by maxval1** [comment]
Additionally, we have `webhooks.populateRelations` set to `true` because of this issue: https://github.com/ComfortablyCoding/strapi-plugin-io/issues/82

**Comment by codyparnell** [comment]
I found the cause of this issue, as i was running into the same problem.  

[I created a pull request with my fix](https://github.com/ComfortablyCoding/strapi-plugin-io/pull/87)

Basically this was an issue that only effected api tokens with an unlimited  duration. The solution was properly querying these tokens.

**Comment by ComfortablyCoding** [comment]
This should be fixed via #87 in v2.0.6

**Comment by foodiecompany** [comment]
Thanks!!
